libs/core/src/CutFlowManager.cpp

```diff
@@ -39,6 +39,35 @@ CutFlowManager::CutFlowManager(shared_ptr<EventReader> eventReader_, shared_ptr<
       currentIndex++;
     }
   }
+  for(auto collectionName : additionalCutFlowCollections){
+    currentCollectionIndex[collectionName] = 0;
+    weightsAfterCollectionCuts[collectionName] = {};
+    existingCollectionCuts[collectionName] = {};
+    inputCollectionContainsInitial[collectionName] = false;
+    
+    string collectionCutFlowName = "CollectionCutFlow_"+collectionName;
+  
+    if (eventReader->inputFile->Get(collectionCutFlowName.c_str())) {
+      info() << "Input file contains " << collectionCutFlowName << " directory - will store existing cutflow in the output." << endl;
+
+      auto sourceDir = (TDirectory *)eventReader->inputFile->Get(collectionCutFlowName.c_str());
+
+      TIter nextKey(sourceDir->GetListOfKeys());
+      TKey *key;
+
+      while ((key = dynamic_cast<TKey *>(nextKey()))) {
+        TObject *obj = key->ReadObj();
+        auto hist = (TH1D *)obj;
+        string cutName = key->GetName();
+        float sumOfWeights = hist->GetBinContent(1);
+        weightsAfterCollectionCuts[collectionName][cutName] = sumOfWeights;
+        if (cutName == "0_initial") inputCollectionContainsInitial[collectionName] = true;
+        existingCollectionCuts[collectionName].push_back(cutName);
+        delete obj;
+        currentIndex++;
```

Comment:
I don't know I'm adding it to different variables for example weightsAfterCuts vs. weightsAfterCollectionCuts[collectionName]

Reply:
For this, see the next comment - I think it could work in a very similar way, handling both cases depending on whether you provide a collection name or not.